src/sentry/auth/helper.py

```diff
@@ -333,6 +334,22 @@ def _get_organization_member(self, auth_identity: AuthIdentity) -> OrganizationM
         except OrganizationMember.DoesNotExist:
             return self._handle_new_membership(auth_identity)
 
+    @staticmethod
+    def _get_user(identity: Identity) -> Optional[User]:
+        email = identity.get("email")
+        if email is None:
+            return None
+
+        # TODO(dcramer): its possible they have multiple accounts and at
+        # least one is managed (per the check below)
+        try:
+            return User.objects.filter(
+                id__in=UserEmail.objects.filter(email__iexact=email).values("user"),
+                is_active=True,
+            ).first()
+        except IndexError:
+            return None
```

Comment:
Can we add _logger.error_?

Reply:
That's not necessary. This exception merely indicates that no `User` exists with the identity's email address, which is an expected condition.